Assets/_Scripts/Samurai/DistanceToStalkpoints.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DistanceToStalkpoints : MonoBehaviour
+{
+    public Camera cam;
+    public GameObject[] stalkpoints;
+    public GameObject closest;
+    public GameObject sClosest;
+    public float sDistance;
+    public float distance=100;
+    private int i = 0;
+    private int ii = 0;
+    private RaycastHit hit;
+
+    private void Update()
```

Comment:
If this ever becomes a problem in efficiency you should put this in a coroutine and give it some time before it updates the distances. Now, I know it's working fine but it still bothers me that it's in the Update function so I had to give my suggestion